include/eve/traits/transparent.hpp

```diff
@@ -0,0 +1,28 @@
+//==================================================================================================
+/*
+	EVE - Expressive Vector Engine
+	Copyright : EVE Project Contributors
+	SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <type_traits>
+
+namespace eve
+{
+	// Default case: normal types do not have specific storage type
+	template <typename T>
+	struct transparent_trait {
+			using type = T;
```

Comment:
what's with formatting.

We don't call them trait in c++.

You call them 

```
struct transparent;

transparent_t
```

I don't like the word transparent very much even if it's used in rust.
Here - let's do vote best of 3 - if you and @jfalcou both think that transparent is the word - fine.

I would go from `equivalent_of`, `identical_to` or smth.

Reply:
Equivalent_tp sounds good to me

Reply:
> If we keep `equivalent_to` for the trait, maybe `has_equivalent` works better for the concept ? 
I agree

> And then `as_equivalent_t` and `as_equivalent` (to better differentiate with `equivalent_to`) for the inner type alias, and the cast function ?

I wish we keep name starting by `as` for the type wrapper familly of type. We can probably juse have `equivalent_to_t`
and some `equivalent_value` function.